### ISSUE TITLE
Add ability to multiple selection when importing files

### ISSUE THREAD

**Issue by Bala621** [synfig/synfig#2403]
i can't import more than one image or file. it is very hard to import multiple files.

if add ability to ctrl+ and shift+ selection. it will be very easy
![importb select](https://user-images.githubusercontent.com/68197434/140599634-a9bc59a2-48b7-4454-8bea-76993a3f430e.jpg)

**Comment by b-aaz** [comment]
There is a workaround (atleast in linux)
Just select your files and drag and drop!

**Comment by nickleus27** [comment]
I am interested in this issue.  Is there any contributing documentation I should read? Or can anyone point me in the right direction as far as files I should be looking at?

**Comment by ice0** [comment]
@nickleus27 Hi!

You can start from here: https://synfig-docs-dev.readthedocs.io/en/latest/

GUI files located in `synfig/synfig-studio` folder.

**Comment by nickleus27** [comment]
@ice0 cool thanks.

**Comment by nickleus27** [comment]
@ice0 Hi how are you today? I think I have found the gtk method for selecting multiple files.  I am new to C++ GUI's and I would need to add it.  After looking through the codebase I am thinking somewhere in synfig/synfig-studio/src/gui/docks/, synfig/synfig-studio/src/gui/trees/, or in the synfig-studio/src/gui/app.cpp file itself?  Would appreciate any advice.

**Comment by rodolforg** [comment]
You can `git grep` to search for `import` :) I think it is on app.cpp. A lot of open dialogs are there.

**Comment by nickleus27** [comment]
@ice0 thank you that is really helpful. right now I am having trouble building from source.  When I run ./2-build-debug.sh after installing all the dependencies I run into problem.  When the build is configuring it exits with error:  You need to install Boost >= 1.53.0.  However, I thought that was installed when I ran the previous ./1-setup-osx-brew.sh?  I am on macos, is there any path that I need to export in my .zshrc file?  I have tried with the path in opt/homebrew/cellar/boost/1.76.0

**Comment by nickleus27** [comment]
@ice0 I fixed my last problem with installing boost from the tar file.  But now the build is exiting with error:

  CXXLD    libmod_openexr.la
ld: warning: ignoring file /usr/local/opt/gettext/lib/libintl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "half::_eLut", referenced from:
      half::half(float) in libmod_openexr_la-trgt_openexr.o
  "half::convert(int)", referenced from:
      half::half(float) in libmod_openexr_la-trgt_openexr.o
  "Imf_2_5::RgbaOutputFile::writePixels(int)", referenced from:
      exr_trgt::end_frame() in libmod_openexr_la-trgt_openexr.o
  "Imf_2_5::RgbaOutputFile::setFrameBuffer(Imf_2_5::Rgba const*, unsigned long, unsigned long)", referenced from:
      exr_trgt::end_frame() in libmod_openexr_la-trgt_openexr.o
  "Imf_2_5::RgbaOutputFile::RgbaOutputFile(char const*, int, int, Imf_2_5::RgbaChannels, float, Imath_2_5::Vec2<float>, float, Imf_2_5::LineOrder, Imf_2_5::Compression, int)", referenced from:
      exr_trgt::start_frame(synfig::ProgressCallback*) in libmod_openexr_la-trgt_openexr.o
  "Imf_2_5::globalThreadCount()", referenced from:
      exr_trgt::start_frame(synfig::ProgressCallback*) in libmod_openexr_la-trgt_openexr.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [libmod_openexr.la] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

At the moment I think solving this might be over my head.  If you have any advice that would be great.  Otherwise I will have to pass on this issue, since I would imagine I should re-build my changes to make sure they are working correctly before submitting a pull request.

**Comment by ice0** [comment]
@nickleus27 
Looks like you are building on M1?
Try CMake build: https://synfig-docs-dev.readthedocs.io/en/latest/building/Building%20using%20CMake.html

**Comment by ice0** [comment]
> @ice0 thank you that is really helpful. right now I am having trouble building from source. When I run ./2-build-debug.sh after installing all the dependencies I run into problem. When the build is configuring it exits with error: You need to install Boost >= 1.53.0. However, I thought that was installed when I ran the previous ./1-setup-osx-brew.sh? I am on macos, is there any path that I need to export in my .zshrc file? I have tried with the path in opt/homebrew/cellar/boost/1.76.0

I'll make a fix soon.

**Comment by nickleus27** [comment]
@ice0 awesome, cmake seems to be working.  Any advice when I want to recompile just the synfig-studio files after I make changes to gui?

**Comment by ice0** [comment]
> @ice0 awesome, cmake seems to be working. Any advice when I want to recompile just the synfig-studio files after I make changes to gui?

Just type `make -j8` in you build directory - CMake detects changes automatically.  
Final binary will be in `<build_directory>/output/Debug/bin/` folder.

**Comment by nickleus27** [comment]
@ice0 for some reason why i tried to recompile with cmake today i got error messages that I am missing packages.  I have been adding PKG_CONFIG_PATH to my .zshrc file for a long time now.  Every time I add the missing package there is a new one.  Any idea how to fix this?

**Comment by ice0** [comment]
@nickleus27 Can you post last 10-20 lines of your error message?

**Comment by ice0** [comment]
Also can you show output of `pkg-config --list-all` command?

**Comment by nickleus27** [comment]
@ice0 Whenever I run make I get an error message about a missing package.  Then I add the package path to .zshrc and then I get a new missing package error.  This is the current missing package error:

-- Checking for module 'gtkmm-3.0'
--   Package 'gdk-pixbuf-2.0', required by 'gdk-3.0', not found
CMake Error at /opt/homebrew/Cellar/cmake/3.19.4/share/cmake/Modules/FindPkgConfig.cmake:553 (message):
  A required package was not found
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.19.4/share/cmake/Modules/FindPkgConfig.cmake:741 (_pkg_check_modules_internal)
  synfig-studio/src/CMakeLists.txt:33 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/Users/nanderson/synfig.git/cmake-build/CMakeFiles/CMakeOutput.log".
See also "/Users/nanderson/synfig.git/cmake-build/CMakeFiles/CMakeError.log".
make: *** [cmake_check_build_system] Error 1

(base) nanderson@Nicholass-MacBook-Air cmake-build % pkg-config --list-all
yaml-0.1              LibYAML - Library to parse and emit YAML
libzmq                libzmq - 0MQ c++ library
cairo-ps              cairo-ps - PostScript surface backend for cairo graphics library
MagickCore-7.Q16HDRI  MagickCore - MagickCore - C API for ImageMagick (ABI Q16HDRI)
xf86vidmodeproto      XF86VidModeProto - XF86VidMode extension headers
MagickWand-7.Q16HDRI  MagickWand - MagickWand - C API for ImageMagick (ABI Q16HDRI)
damageproto           DamageProto - Damage extension headers
pangoot               Pango OT - OpenType font support for Pango (deprecated)
dpmsproto             DPMSProto - DPMS extension headers
ncurses++             ncurses++ - ncurses 6.2 add-on library
kbproto               KBProto - KB extension headers
gio-unix-2.0          GIO unix specific APIs - unix specific headers for glib I/O library
libffi                libffi - Library supporting Foreign Function Interfaces
ncursesw              ncursesw - ncurses 6.2 library
cairo-png             cairo-png - PNG functions for cairo graphics library
dmxproto              DMXProto - DMX extension headers
liblzma               liblzma - General purpose data compression library
cairo-script          cairo-script - script surface backend for cairo graphics library
sqlite3               SQLite - SQL database engine
glibmm-2.68           glibmm - C++ wrapper for GLib
fftw3                 FFTW - fast Fourier transform library
xext                  Xext - Misc X Extension Library
gdk-3.0               GDK - GTK+ Drawing Kit
glibmm-2.4            glibmm - C++ wrapper for GLib
sigc++-3.0            libsigc++ - Typesafe signal and callback system for C++
x11                   X11 - X Library
ncurses               ncurses - ncurses 6.2 library
tcl                   Tool Command Language - Tcl is a powerful, easy-to-learn dynamic programming language, suitable for a wide range of uses.
ncurses++w            ncurses++w - ncurses 6.2 add-on library
xcb-shape             XCB Shape - XCB Shape Extension
libcrypto             OpenSSL-libcrypto - OpenSSL cryptography library
libxml++-2.6          libxml++ - C++ wrapper for libxml
xcb-dri2              XCB DRI2 - XCB DRI2 Extension
tk                    The Tk Toolkit - Tk is a cross-platform graphical user interface toolkit, the standard GUI not only for Tcl, but for many other dynamic languages as well.
xcb-composite         XCB Composite - XCB Composite Extension
gtk+-quartz-3.0       GTK+ - GTK+ Graphical UI Library
apr-util-1            APR Utils - Companion library for APR
Magick++              Magick++ - Magick++ - C++ API for ImageMagick (ABI Q16HDRI)
harfbuzz              harfbuzz - HarfBuzz text shaping library
pixman-1              Pixman - The pixman library (version 1)
xcb-dri3              XCB DRI3 - XCB DRI3 Extension
pangofc               Pango FC - Fontconfig support for Pango
cairo-quartz-font     cairo-quartz-font - Quartz font backend for cairo graphics library
pthread-stubs         pthread stubs - Meta package for pthread symbols - defaults to heavyweight ones if the C runtime does not provide lightweight ones.
panelw                panelw - ncurses 6.2 add-on library
expat                 expat - expat XML parser
pangocairo            Pango Cairo - Cairo rendering support for Pango
xdmcp                 Xdmcp - X Display Manager Control Protocol library
libpcreposix          libpcreposix - PCREPosix - Posix compatible interface to libpcre
glib-2.0              GLib - C Utility Library
fftw3f                FFTW - fast Fourier transform library
menuw                 menuw - ncurses 6.2 add-on library
xcb-xtest             XCB XTEST - XCB XTEST Extension
presentproto          PresentProto - Present extension headers
ImageMagick           ImageMagick - ImageMagick - convert, edit, and compose images (ABI Q16HDRI)
gdkmm-3.0             gdkmm - C++ binding for the GDK drawing kit
inputproto            InputProto - Input extension headers
libssl                OpenSSL-libssl - Secure Sockets Layer and cryptography libraries
xcb-shm               XCB Shm - XCB Shm Extension
xcb-dpms              XCB DPMS - XCB DPMS Extension
libpng16              libpng - Loads and saves PNG files
sigc++-2.0            libsigc++ - Typesafe signal and callback system for C++
bigreqsproto          BigReqsProto - BigReqs extension headers
xextproto             XExtProto - XExt extension headers
Imath                 Imath - Imath library: vector/matrix and math operations, plus the half type.
xcb-xselinux          XCB SELinux - XCB SELinux Extension
tinfo                 tinfo - ncurses 6.2 terminal interface library
gio-2.0               GIO - glib I/O library
fontconfig            Fontconfig - Font configuration and customization library
cairo-quartz          cairo-quartz - Quartz surface backend for cairo graphics library
menu                  menu - ncurses 6.2 add-on library
xf86dgaproto          XF86DGAProto - XF86DGA extension headers
dri3proto             DRI3Proto - DRI3 extension headers
x11-xcb               X11 XCB - X Library XCB interface
libpcre               libpcre - PCRE - Perl compatible regular expressions C library with 8 bit character support
readline              Readline - Gnu Readline library for command line editing
formw                 formw - ncurses 6.2 add-on library
xcb-ge                XCB GenericEvent - XCB GenericEvent Extension
xau                   Xau - X authorization file management libary
tinfow                tinfow - ncurses 6.2 terminal interface library
xcb-xprint            XCB Xprint - XCB Xprint Extension
graphite2             Graphite2 - Font rendering engine for Complex Scripts
cairo-pdf             cairo-pdf - PDF surface backend for cairo graphics library
resourceproto         ResourceProto - Resource extension headers
gtk+-3.0              GTK+ - GTK+ Graphical UI Library
zlib                  zlib - zlib compression library
MagickWand            MagickWand - MagickWand - C API for ImageMagick (ABI Q16HDRI)
fontsproto            FontsProto - Fonts extension headers
gmodule-2.0           GModule - Dynamic module loader for GLib
libmng                libmng - Loads PNG/MNG/JNG files
xcb-sync              XCB Sync - XCB Sync Extension
xcb-res               XCB Res - XCB X-Resource Extension
cairo-ft              cairo-ft - FreeType font backend for cairo graphics library
fftw3l                FFTW - fast Fourier transform library
fribidi               GNU FriBidi - Unicode Bidirectional Algorithm Library
cairo-xlib-xrender    cairo-xlib-xrender - Xlib Xrender surface backend for cairo graphics library
glproto               GLProto - GL extension headers
videoproto            VideoProto - Video extension headers
xcb-xinput            XCB XInput - XCB XInput Extension (EXPERIMENTAL)
harfbuzz-subset       harfbuzz-subset - HarfBuzz font subsetter
python3-embed         Python - Embed Python into an application
xcb-xvmc              XCB XvMC - XCB XvMC Extension
xcb                   XCB - X-protocol C Binding
giomm-2.4             giomm - C++ binding for gio
libjpeg               libjpeg - Reads and writes JPEG files
Magick++-7.Q16HDRI    Magick++ - Magick++ - C++ API for ImageMagick (ABI Q16HDRI)
xcmiscproto           XCMiscProto - XCMisc extension headers
xcb-xevie             XCB Xevie - XCB Xevie Extension
recordproto           RecordProto - Record extension headers
xcb-record            XCB Record - XCB Record Extension
xcb-randr             XCB RandR - XCB RandR Extension
xcb-xfixes            XCB XFixes - XCB XFixes Extension
gmodule-no-export-2.0 GModule - Dynamic module loader for GLib
xineramaproto         XineramaProto - Xinerama extension headers
gtk+-unix-print-3.0   GTK+ - GTK+ Unix print support
dri2proto             DRI2Proto - DRI2 extension headers
python3               Python - Build a C extension for Python
libiodbc              iODBC - iODBC Driver Manager
xcb-render            XCB Render - XCB Render Extension
xcb-damage            XCB Damage - XCB Damage Extension
xproto                Xproto - Xproto headers
scrnsaverproto        ScrnSaverProto - ScrnSaver extension headers
cairo-gobject         cairo-gobject - gobject functions for cairo graphics library
cairo-xcb-shm         cairo-xcb-shm - XCB/SHM functions for cairo graphics library
gtkmm-3.0             gtkmm - C++ binding for the GTK+ toolkit
renderproto           RenderProto - Render extension headers
xcb-xinerama          XCB Xinerama - XCB Xinerama Extension
cairo-xcb             cairo-xcb - XCB surface backend for cairo graphics library
OpenEXR               OpenEXR - OpenEXR image library
gail-3.0              Gail - GNOME Accessibility Implementation Library
libpng                libpng - Loads and saves PNG files
gobject-2.0           GObject - GLib Type, Object, Parameter and Signal Library
ImageMagick-7.Q16HDRI ImageMagick - ImageMagick - convert, edit, and compose images (ABI Q16HDRI)
xrender               Xrender - X Render Library
cairo-xlib            cairo-xlib - Xlib surface backend for cairo graphics library
xf86bigfontproto      XF86BigFontProto - XF86BigFont extension headers
gthread-2.0           GThread - Thread support for GLib
xcb-screensaver       XCB Screensaver - XCB Screensaver Extension
apr-1                 APR - The Apache Portable Runtime library
cairo-svg             cairo-svg - SVG surface backend for cairo graphics library
pangoft2              Pango FT2 and Pango Fc - Freetype 2.0 and fontconfig font support for Pango
applewmproto          AppleWMProto - AppleWM extension headers
harfbuzz-icu          harfbuzz-icu - HarfBuzz text shaping library ICU integration
randrproto            RandrProto - Randr extension headers
compositeproto        CompositeExt - Composite extension headers
form                  form - ncurses 6.2 add-on library
libxml-2.0            libXML - libXML library version2.
xcb-present           XCB Present - XCB Present Extension
cairo-fc              cairo-fc - Fontconfig font backend for cairo graphics library
cairo-tee             cairo-tee - tee surface backend for cairo graphics library
xcb-xv                XCB Xv - XCB Xv Extension
openssl               OpenSSL - Secure Sockets Layer and cryptography libraries and tools
python-3.9            Python - Build a C extension for Python
xcb-xkb               XCB XKB - XCB Keyboard Extension (EXPERIMENTAL)
xf86driproto          XF86DRIProto - XF86DRI extension headers
libpcrecpp            libpcrecpp - PCRECPP - C++ wrapper for PCRE
libsodium             libsodium - A modern and easy-to-use crypto library
harfbuzz-gobject      harfbuzz-gobject - HarfBuzz text shaping library GObject integration
gmodule-export-2.0    GModule - Dynamic module loader for GLib
panel                 panel - ncurses 6.2 add-on library
python-3.9-embed      Python - Embed Python into an application
pango                 Pango - Internationalized text handling
xcb-xf86dri           XCB XFree86-DRI - XCB XFree86-DRI Extension
gdk-quartz-3.0        GDK - GTK+ Drawing Kit
xcb-glx               XCB GLX - XCB GLX Extension
giomm-2.68            giomm - C++ binding for gio
freetype2             FreeType 2 - A free, high-quality, and portable font engine.
cairo                 cairo - Multi-platform 2D graphics library
MagickCore            MagickCore - MagickCore - C API for ImageMagick (ABI Q16HDRI)
fixesproto            FixesProto - X Fixes extension headers
cairo-quartz-image    cairo-quartz-image - Quartz Image surface backend for cairo graphics library

**Comment by nickleus27** [comment]
> > but attempting to link with file built for macOS-x86_64
> 
> Do you have both versions of Homebrew installed? I mean x64 and arm?

I may have both installed. I cant remember. How can I check. brew -v outputs this:

(base) nanderson@Nicholass-MacBook-Air cmake-build % brew -v
Homebrew 3.3.4-72-g27a7629
Homebrew/homebrew-core (git revision 1b2746fd0b7; last commit 2021-11-20)
Homebrew/homebrew-cask (git revision fbb42995b4; last commit 2021-11-20)

When I first got the M1 Mac no one had M1 native software.  After a while of having the M1 I started re-installing M1 native versions as they came out.  I think I did this with homebrew.  How do I check for the two versions?

Really appreciate your help

**Comment by nickleus27** [comment]
if have brew at /opt/homebrew and i also found it here /usr/local/Homebrew
Could this be a problem?

**Comment by ice0** [comment]
> if have brew at /opt/homebrew and i also found it here /usr/local/Homebrew Could this be a problem?

Yes. I haven't tried using both at the same time, but in theory it could lead to some problems. 
`/opt/homebrew` - is arm version
`/usr/local/Homebrew` - is x86_64 version

You can check you current chosen architecture by typing `arch` in console.
For example this is my output:
```
$ arch
arm64
```

> Then I add the package path to .zshrc

This also can lead to problem. Better to remove it (as it can pick the wrong folder). Currently you have 2 versions (arm64/x64) of `pkg-config` installed (which we use to get the library paths), so adding the paths manually can lead to some strange errors. 

First try deleting the build folder and re-create it again. CMake does not update library paths after upgrading the package (for example, after the `brew upgrade` command).

After that check `brew doctor` output. If no error output here, then try to re-run `1-setup-osx-brew.sh` again. 

If this does not help (we are not reinstalling already installed packages), try installing the required packages manually. 

For example there is no `gdk-pixbuf-2.0` in your `pkg-config --list-all` output, so it need to be installed.

**Comment by nickleus27** [comment]
I deleted build folder and deleted the paths in the .zshrc, and this did the trick. Thank you @ice0 .

**Comment by nickleus27** [comment]
@ice0 I was able to get the gui to select multiple files.  However, I have not figured out how to get them all to import when you press "import" button.  I made my change in App::dialog_open_file() and it seems to get called in CanvasView::import_file().  Can I do a for loop somewhere for file row selected?

**Comment by ice0** [comment]
@nickleus27 
As you may have noticed, `App::dialog_open_file` returns the selected filename using a reference variable. 
Therefore, I propose to divide this task into smaller ones. 
1. Allow the `dialog_open_file` method to return multiple files:
a) Change `std::string &filename` parameter to `std::vector<std::string>& filenames` parameter.
b) Add a new `bool allow_multiple_selection` parameter to make this method backward compatible. In case the calling function does not need multiple files. 
c) Fix calling functions/methods and `dialog_open_file` method.
d) At this step you will have `App::dialog_open_file(const std::string &title, std::vector<std::string>& filenames, std::string preference, bool allow_multiple_selection)` method.
e) Make PR with this changes.

2. Make a small refactor of `CanvasView::import_file()` method:
a) Extract `is_same_file` check to function (to make code cleaner)
b) Make PR with this changes.

3. Final changes:
a) Decide if all imported files should be selected on `Layers` panel after import.
b) Add the ability to import multiple files.
c) Make PR with this changes.

**Comment by nickleus27** [comment]
@ice0 Thanks, you're awesome.  I really appreciate you helping me along.  Well I was on the right track because I have already done 1a.  But your advice for 1b i didn't think of and that is a great idea.  Thank you so much, I will work on these steps.

**Comment by nickleus27** [comment]
@ice0 Alright, I have almost finished step 1 from your comment earlier.  I will review changes before opening a pull request tomorrow.  The only thing is I overloaded the dialog_open_file function instead of creating the bool allow_multiple_selection parameter.  So now there is App::dialog_open_file(const std::string &title, std::vector<std::string> &filenames, std::string preference) and the original App::dialog_open_file(const std::string &title, std::string &filename, std::string preference).  The changes compile and work as far as importing multiple files.  At least it is working for me.  Do you think that will be good, or can you explain a little more about how to use the bool parameter.  Thanks for your help.

**Comment by ice0** [comment]
I suggested using an additional parameter because this avoids copying the code, making it easier to maintain. 
For example, if we copy the code of this function into a new one, then when adding a new file type, we will need to remember to add it to both functions. 
I mean this part:
```
	// 2.1 Image files
	Glib::RefPtr<Gtk::FileFilter> filter_image = Gtk::FileFilter::create();
	filter_image->set_name(_("Images (*.png, *.jpeg, *.bmp, *.svg)"));
	filter_image->add_mime_type("image/png");
```

Another approach is also possible. You can extract the function code into a function with a parameter, and set this parameter, for example: 
```
bool App::dialog_open_file_ext(const std::string &title, std::vector<std::string> &filenames, std::string preference, bool allow_multiple_selection) {
...
}

bool App::dialog_open_file(const std::string &title, std::string &filename, std::string preference) {
  std::vector<std::string> filenames> filenames;
  if (dialog_open_file_ext(title, filenames, preference, false)) {
    filename = filenames.front();
    return true;
  }
  return false;
}

bool App::dialog_open_file(const std::string &title, std::vector<std::string>& filenames>, std::string preference) {
  return dialog_open_file_ext(title, filenames, preference, true);
}

```


But it seems to me that it is more complicated because we added an additional method (which can be confusing to the user), plus this method is only used in two places, so it is easier to fix them in place. 


> I have not made changes to #ifdef USE_WIN32_FILE_DIALOGS in dialog_open_file function because I am not sure how to test to see if the changes work since I am not on a win32 platform?

That's ok. We are not using this feature now, so we can fix it later.

**Comment by nickleus27** [comment]
@ice0 fyi: I pulled from the recent master today and I can no long build from cmake.  i get this error:

[ 45%] Linking CXX shared module ../../../../output/Debug/lib/synfig/modules/libmod_mng.so
ld: library not found for -lmng
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [output/Debug/lib/synfig/modules/libmod_mng.so] Error 1
make[1]: *** [synfig-core/src/modules/mod_mng/CMakeFiles/mod_mng.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

**Comment by ice0** [comment]
@nickleus27 
Fixed in https://github.com/synfig/synfig/pull/2441
(Currently not merged yet, so you need to copy the changes to yourself)

**Comment by nickleus27** [comment]
@ice0 thanks.

**Comment by nickleus27** [comment]
@ice0 fyi i tried re cloning synfig just now and i get this error when I use cmake ..

```
-- Checking for module 'sigc++-2.0'
--   No package 'sigc++-2.0' found
CMake Error at /opt/homebrew/Cellar/cmake/3.19.4/share/cmake/Modules/FindPkgConfig.cmake:553 (message):
  A required package was not found
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.19.4/share/cmake/Modules/FindPkgConfig.cmake:741 (_pkg_check_modules_internal)
  synfig-core/src/CMakeLists.txt:143 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/Users/nanderson/synfig/cmake-build/CMakeFiles/CMakeOutput.log".
See also "/Users/nanderson/synfig/cmake-build/CMakeFiles/CMakeError.log".

``

I don't know very much about cmake so I am not sure how to debug, but thought you might want to know in case there is a bug.

**Comment by ice0** [comment]
> I don't know very much about cmake so I am not sure how to debug, but thought you might want to know in case there is a bug

We use `pkg-config` to get the package paths. So you can just type:
```
pkg-config --list-all | grep -i sigc
```
to check if sigc++-2.0 is installed.

**Comment by nickleus27** [comment]
@ice0 thank you.  I should have done that.  I just assumed it was there because I already installed it.  For some reason I had to source ~/.zshrc and then it found it.  It is weird because I did not make any changes to ~/.zshrc.

**Comment by ice0** [comment]
@nickleus27 

> for the refactor of CanvasView::import_file() method should i make a new branch and submit just these changes as a pr

Yes. If you need current changes, you can create a new branch from the current branch and continue here. If not - just create a new branch from master.

**Comment by rodolforg** [comment]
fixed by #2451
